src/rust/bitbox02-rust/src/bb02_async.rs

```diff
@@ -48,8 +48,8 @@ impl<O> core::future::Future for AsyncOption<'_, O> {
 
 /// Waits for an option to contain a value and returns that value, leaving `None` in its place.
 /// E.g. `assert_eq!(option(&Some(42)).await, 42)`.
-pub fn option<'a, O>(option: &'a RefCell<Option<O>>) -> AsyncOption<'a, O> {
-    AsyncOption(&option)
+pub fn option<O>(option: &RefCell<Option<O>>) -> AsyncOption<'_, O> {
```

Comment:
What's the reason for `'_` in returned value? I tried without it. rust seems to be able to figure it out without issues and clippy doesn't complain.

Reply:
I might have copied it from the original clippy fix suggestion?

I am surprised it works without it, after all, it is an explicit param in the declaration of AsyncOption...

Anyway, removed.